.ci/ci_cache_components.sh

```diff
@@ -0,0 +1,215 @@
+	tar -cJf "${getinfo}.tar.xz" "${getinfo}"
+	sha256sum "${getinfo}.tar.xz" > "${image_name}-tarball.sha256sum"
+	sha256sum -c "${image_name}-tarball.sha256sum"
+	rm "${getinfo}"
+}
+
+# This function receives as arguments the component name (qemu, nemu, kernel, etc)
+# as well as the version
+build_cache() {
+	local component_name="$1"
+	local component_version="$2"
+	local check_image=$(echo "${component_version}" | grep kata || true)
+	local check_nemu=$(echo "${component_name}" | grep nemu || true)
+	local check_qemu=$(echo "${component_name}" | grep qemu || true)
+	local check_kernel=$(echo "${component_name}" | grep vmlinu || true)
+	local check_initrd=$(echo "${component_name}" | grep initrd || true)
+
+	# Verify which kind of image we are using (initrd or not)
+	if [ ! -z "${check_image}" ]; then
+		if [ ! -z "${check_initrd}" ]; then
+			image_path="${kata_dir}/kata-containers-initrd.img"
+			image_name="initrd"
+		else
+			image_path="${kata_dir}/kata-containers.img"
+			image_name="image"
+		fi
+		path=$(readlink -f "${image_path}")
+		echo $(basename "${path}") > "latest-${image_name}"
+		sudo cp "${path}" "${kata_dir}/osbuilder-${image_name}.yaml"  .
+	else
+		echo "${component_version}" >  "latest"
+	fi
+
+	# In the case of nemu and qemu we have the tar file in
+	# a specific location /tmp
+	if [ ! -z "${check_nemu}" ] || [ ! -z "${check_qemu}" ]; then
+		cp -a "/tmp/${component_name}" .
+	fi
+
+	# In the case of the kernel we have it in the kata dir
+	if [ ! -z "${check_kernel}" ]; then
+		cp "${kata_dir}/${component_name}" .
+	fi
+
+	sudo chown -R "${USER}:${USER}" .
+
+	if [ ! -z "${check_image}" ]; then
+		sha256sum "$(cat latest-${image_name})" > "sha256sum-${image_name}"
+        	sha256sum -c "sha256sum-${image_name}"
+	elif [ ! -z "${check_kernel}" ]; then
+		sha256sum "${component_name}" >> "sha256sum-kernel"
+		cat sha256sum-kernel
+	else
+		sha256sum "${component_name}" > "sha256sum-${component_name}"
+		cat "sha256sum-${component_name}"
+	fi
+}
+
+usage() {
+cat <<EOT
+Usage: $script_name [options]
+Description: This script builds the cache of several Kata components.
+Options:
+
+-i      Run image cache
+-k      Run kernel cache
+-n      Run nemu cache
+-q      Run qemu cache
+-r      Run image initrd cache
+-h      Shows help
+
+EOT
+}
+
+main() {
+	local OPTIND
+	while getopts ":iknqrh:" opt; do
+		case "$opt" in
+		i)
+			build_image="1"
+			;;
+		k)
+			build_kernel="1"
+			;;
+		n)
+			build_nemu="1"
+			;;
+		q)
+			build_qemu="1"
+			;;
+		r)
+			build_image_initrd="1"
+			;;
+		h)
+			usage
+			exit 0;
+			;;
+		esac
+	done
+	shift $((OPTIND-1))
+
+	[[ -z "$build_kernel" ]] && \
+	[[ -z "$build_qemu" ]] && \
+	[[ -z "$build_nemu" ]] && \
+	[[ -z "$build_image" ]] && \
+	[[ -z "$build_image_initrd" ]] && \
+		help && die "Must choose at least one test"
+
+	mkdir -p "${WORKSPACE}/artifacts"
+ 	pushd "${WORKSPACE}/artifacts"
+	echo "artifacts:"
+
+	[ "$build_kernel" == "1" ] && cache_kernel_artifacts
```

Comment:
Nit: Recently I started to use true and false strings for this kind of validation.